internal/pkg/deploy/upload/asset/asset.go

```diff
@@ -140,30 +143,40 @@ func (u *ArtifactBucketUploader) uploadAssets(assets []asset) error {
 	return g.Wait()
 }
 
-// uploadAssetMappingFile uploads a JSON file to u.AssetMappingPath containing
+// uploadAssetMappingFile uploads a JSON file to u.AssetMappingDir containing
 // the current location of each file in the artifact bucket and the desired location
 // of the file in the destination bucket. It has the format:
 //
-//	{
+//	[{
 //	  "path": "local-assets/12345asdf",
 //	  "destPath": "index.html"
-//	}
-func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) error {
-	// stable output
+//	}]
+//
+// The uploaded path of the file is returned along with an error, if any.
+func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) (string, error) {
 	sort.Slice(assets, func(i, j int) bool {
 		if assets[i].ArtifactBucketPath != assets[j].ArtifactBucketPath {
 			return assets[i].ArtifactBucketPath < assets[j].ArtifactBucketPath
 		}
 		return assets[i].ServiceBucketPath < assets[j].ServiceBucketPath
 	})
 
+	// hash using the sorted list so order-only changes in assets (caused by
+	// manifest or walkFn reordering) doesn't result in a mapping name change.
+	hash := sha256.New()
+	for _, asset := range assets {
```

Comment:
Very nit just a note that i don't think `asset.localPath` matters

Reply:
chatted offline, we are only hashing content